### ISSUE TITLE
Decouple the homebridge plugin from the API implementation

### ISSUE THREAD

**Issue by koush** [hjdhjd/homebridge-unifi-protect#537]
First off, great work on this API. I ended up replacing my own implementation with yours with ease. This leads me to the problem:

**Is your feature request related to a problem? Please describe:**
I would like to use the Protect API bindings outside of the context of Homebridge. That means decoupling your API implementation from Homebridge.

**Describe the solution you'd like:**
I'd like a decoupled API binding that has no dependencies on Homebridge. In fact, I've done that:
https://github.com/koush/unifi-protect

I have not published this yet, as I wanted you to retain ownership of the repository and project, since it is all your code. I've simply split the project into two discrete parts. unifi-protect which provides a node api. And this homebridge-unifi-protect would use that API implementation to integrate with Homebridge.

There were very few changes. I only had to change how the logging and debug functions were passed.

**Describe alternatives you've considered:**
There is no alternative.

**Comment by hjdhjd** [comment]
Thanks for the note and the effort...I designed the API implementation to be intentionally easy to decouple for easier reuse.

Decoupling is on my backlog given the number of requests I've received on this and will publish it in the coming weeks / months...stay tuned! There's one more component I need to finish engineering.

**Comment by koush** [comment]
No sweat, I went ahead and did it myself. The repository is published at https://github.com/koush/unifi-protect and on npm "unifi-protect".

**Comment by hjdhjd** [comment]
Cool. Do you intend to revert ownership of the NPM package, or is it your intention to fork my codebase? My concern is that I would really prefer to keep a single codebase and this NPM is going to create inevitable confusion and rather than focus efforts.

**Comment by koush** [comment]
I'll happily turn over ownership to you once you are ready to take over maintenance.

**Comment by hjdhjd** [comment]
Send it over...it's a good prod for a couple of other things on my plate as well. Thanks!

**Comment by koush** [comment]
You've been added. All I ask is that you leave me on the maintainer access list until your first release! I understand you're busy, and if any issues come up, I'd like to be able to push updates myself until you're on top of it again.

**Comment by hjdhjd** [comment]
Will do. Thanks...and I see you're interested in HKSV. It's ugly...but if you make some headway, Protect's the natural first place to implement it.

**Comment by koush** [comment]
HSKV is done. Just testing now! Works well. I'm not on homebridge though. Just using a forked HAP-NodeJS directly in my own project. Will be contributing those changes back soon (but there's a lot of additional plumbing necessary in Homebridge and every plugin).

Fairly easy to set up if you want to give it a shot:

```sh
git clone https://github.com/koush/scrypted
cd scrypted/server
npm install
npm run serve
```

Then install the Scrypted UniFi plugin. You may also want the Prebuffer plugin, it retains a video loop for HSKV that can be quickly served up on motion. It's a soft requirement by spec (otherwise it misses the lead up to the motion).

<img width="1592" alt="image" src="https://user-images.githubusercontent.com/73924/131205454-91b51203-5a5a-4981-99c2-52ccb24e7efa.png">

**Comment by hjdhjd** [comment]
I'll poke at it...we'd love the contributions back in Homebridge. 

Tagging some partners in crime who will find this one interesting too...@Supereg  @oznu @Sunoo - take a peek...

**Comment by koush** [comment]
Yep I've already been chatting with @supereg, I read his spec and implemented it. My stuff is all open source including the HAP fork. As mentioned, I'll submit a pull request once I'm confident everything is working properly. Been using it for a few weeks and it seems rock solid. I did notice Apple TV is not a great home hub and needs restarting, while the HomePods seem to be more reliable.

Homebridge is great, but like working on stuff so I built Scrypted. Main thing I wanted was a platform that I could launch/debug plugins while it was running. No restarts!
